articles/execute_process_refactor.md

```diff
@@ -0,0 +1,509 @@
+|cwd|the directory in which to run the executable|
+|env|dictionary of environment variables to be used, starting from a clean environment. If `None`, the current environment is used.|
+|additional\_env|dictionary of environment variables to be added. If `env` was `None`, they are added to the current environment. If not, `env` is updated with `additional_env`.|
+
+#### Properties
+
+Accessors would be provided for the various constructor parameters: `cmd`, `name`, `cwd`, `env`, and `additional_env`.
+
+#### Methods
+
+In the current classes, substitutions are handled through internal method calls. To support better isolation of class functionality, substitutions will be performed in this class, but exposed from a public method which can be overridden by subclasses to perform additional appropriate substitutions.
+
+| Name          | Description                                                  |
+| ------------- | ------------------------------------------------------------ |
+| apply_context | Takes a given LaunchContext and performs actions necessary to prepare to execute the defined command in that context. This will primarily consist of expanding various substitutions. |
+
+#### Events
+
+No events would be handled nor emitted.
+
+### launch_ros.descriptions.Node
+
+This class would represent only the information required to define a node which might be run. It would inherit from `launch.descriptions.Process`, and populate its fields in the same manner as the current `launch_ros.actions.Node`.
+
+#### Constructor
+
+These parameters are drawn from the current `launch_ros.actions.Node`.
+
+|Argument|Description|
+|---|---|
+|node_executable|the name of the executable to find if a package is provided or otherwise a path to the executable to run.|
+|package|name of the ROS package the node executable lives in|
+|node_name| name the node should have                                    |
+|node_namespace|namespace the node should create topics/services/etc in|
+|parameters| list of either paths to yaml files or dictionaries of parameters |
+|remappings|list of from/to pairs for remapping names|
+|arguments|container specific arguments to be passed to the node|
+
+#### Properties
+
+Accessors would be provided for the various constructor parameters: `node_executable`, `package`, `node_name`, `node_namespace`, `parameters`, `remappings`, and `arguments`. Inherited accessors would also be available: `cmd`, `name`, `cwd`, `env`, and `additional_env`.
+
+#### Methods
+
+This class would override the `apply_context` method of `launch.descriptions.Process` to apply additional ROS-specific substitutions, then invoke the superclass method itself. This function would match that of the current `launch_ros.actions.Node` internal method `_perform_substitutions`.
+
+#### Events
+
+No events would be handled or emitted.
+
+### launch_ros.descriptions.ComposableNode
+
+This class would be modified to inherit from `launch_ros.descriptions.Node`. 
```

Comment:
I don't think this is possible, as for example `node_executable` argument doesn't apply for a composable node.
Maybe, both descriptions can inherit from a common base description.